examples/stable/test/property/ic_api/json/test/tests.ts

```diff
@@ -11,177 +11,81 @@ import fc from 'fast-check';
-    largeDict[`key${i}`] = `value${i}`;
-}
-testCases.push({
-    name: 'Dictionary with 20 keys',
-    value: largeDict
-});
-
-// Create a simpler object that doesn't include undefined
-const complexObject = {
-    string: 'test',
-    number: 42,
-    boolean: true,
-    null: null,
-    nan: NaN,
-    infinity: Infinity,
-    negativeInfinity: -Infinity,
-    negativeZero: -0,
-    bigint: BigInt('12345678901234567890'),
-    array: [1, 'test', true],
-    object: { nested: 'value' },
-    typedArray: new Uint8Array([1, 2, 3]),
-    map: new Map([['key', 'value']]),
-    set: new Set(['value'])
-};
-testCases.push({
-    name: 'Complex object with multiple types',
-    value: complexObject
-});
+// Helper function to create a Principal arbitrary
+const principalArb = fc
+    .array(fc.integer({ min: 0, max: 255 }), { minLength: 1, maxLength: 29 })
+    .map((bytes) => Principal.fromUint8Array(Uint8Array.from(bytes)));
+
+// Create an arbitrary for all the special types that jsonStringify can handle
+const specialValueArb = fc.oneof(
+    fc.constant(undefined),
+    fc.bigInt({ min: -9007199254740991n, max: 9007199254740991n }),
+    principalArb,
+    fc.constant(NaN),
+    fc.constant(Infinity),
+    fc.constant(-Infinity),
+    fc.constant(-0),
+    fc.uint8Array({ minLength: 0, maxLength: 10 }),
+    fc.uint16Array({ minLength: 0, maxLength: 10 }),
+    fc.uint32Array({ minLength: 0, maxLength: 10 }),
+    fc.int8Array({ minLength: 0, maxLength: 10 }),
+    fc.int16Array({ minLength: 0, maxLength: 10 }),
+    fc.int32Array({ minLength: 0, maxLength: 10 }),
+    fc.float32Array({ minLength: 0, maxLength: 10 }),
+    fc.float64Array({ minLength: 0, maxLength: 10 })
+);
 
-// Simple object arbitraries for property testing
+// Create safe BigInt array arbitraries with appropriate ranges
+const bigInt64ArrayArb = fc
+    .array(fc.bigInt({ min: -9007199254740991n, max: 9007199254740991n }), {
+        minLength: 0,
+        maxLength: 5
+    })
+    .map((arr) => new BigInt64Array(arr));
+
+const bigUint64ArrayArb = fc
+    .array(fc.bigInt({ min: 0n, max: 9007199254740991n }), {
+        minLength: 0,
+        maxLength: 5
+    })
+    .map((arr) => new BigUint64Array(arr));
+
+// Create an arbitrary for JSON-compatible values (strings, numbers, booleans, arrays, objects)
 const simpleValueArb = fc.oneof(
     fc.string(),
+    fc.float(),
     fc.integer(),
     fc.boolean(),
     fc.constant(null)
 );
 
-// Simple array and object arbitraries
-const simpleArrayArb = fc.array(simpleValueArb, { maxLength: 5 });
+const simpleArrayArb = fc.array(simpleValueArb, { maxLength: 3 });
 const simpleObjectArb = fc.dictionary(fc.string(), simpleValueArb, {
     maxKeys: 5
 });
 
-// Combine them for property testing
-const simpleTestArb = fc.oneof(simpleValueArb, simpleArrayArb, simpleObjectArb);
+// Create a map with simple key/value pairs
+const mapArb = fc
+    .array(fc.tuple(fc.string(), simpleValueArb), { maxLength: 5 })
```

Comment:
Do not use the simpleValueArb here, use the mixedObjectArb. I want all nested structures to have the full set of arbitrary values available.